src/app/server/Server.cpp

```diff
@@ -316,24 +314,23 @@ static CHIP_ERROR OpenPairingWindowUsingVerifier(uint16_t discriminator, PASEVer
     return gRendezvousServer.WaitForPairing(std::move(params), &gTransports, &gSessions, adminInfo);
 }
 
-class ServerCallback : public SecureSessionMgrDelegate
+class ServerCallback : public ExchangeDelegate
 {
 public:
-    void OnMessageReceived(const PacketHeader & header, const PayloadHeader & payloadHeader, SecureSessionHandle session,
-                           System::PacketBufferHandle buffer, SecureSessionMgr * mgr) override
+    void OnMessageReceived(Messaging::ExchangeContext * exchangeContext, const PacketHeader & packetHeader,
+                           const PayloadHeader & payloadHeader, System::PacketBufferHandle buffer) override
```

Comment:
I strongly suggest removing `packetHeader` and `payloadHeader` arguments.

`packetHeader` is used to retrieve node id, but we can get node id from `ExchangeContext` which is more reliable.

`payloadHeader` is only used to get protocol id of the packet, I think we can use protocol id and to replace payloadHeader.

Reply:
This function prototype is to implement the callbacks of ExchangeDelegate interface,  changing this interface will impact all ExchangeContext clients. 

If I understand correctly, you are talking about refactor current ExchangeDelegate interface in general, right? I suggest we register a separate ticket to discuss this